### ISSUE TITLE
Prefab not loaded nor build with scene

### ISSUE THREAD

**Issue by EmptyKeys** [SiliconStudio/xenko#469]
If you use prefab in scene, that prefab is actually not build with the scene and does not have any Referencers. Anyone can remove that prefab and there is not warning etc. On top of that you can't use that prefab in code.

The only solution is to mark that prefab "build as root", which is terrible, because in that case everything is loaded on start of the game and does not solve "no referencers issue" in GameStudio.

**Comment by Benlitz** [comment]
Hi,

It is actually intended that the prefab is not build when used in a scene - the link from entities placed in a scene to their origin prefab is not kept at compilation.
Prefabs are both a design-time tool (to make reusable things that share properties) and a run-time object (usually used to "spawn" entities in your scene). When used for design-time purpose, there is no reason to build them, and we assumed that in most of the case prefabs would be used in this scenario.

Now, in order to have a prefab loadable from a script (so probably to "spawn" it in your scene), marking it as "build as root" is actually the normal workflow. "Build as root" means "compile this asset even if it's not directly referenced from the default scene." - which is the case here. However it doesn't mean that it will be loaded at the start of the game - it is actually not the case, you still need to manually call Content.Load("MyPrefab").

On the other hand, the current indicators of references between assets in the Game Studio are a bit inaccurate and we intend to rework them in the future. They don't make a clear distinction between compile-time dependencies and actual run-time references - in addition to a few bugs when sometimes a reference is improperly displayed.

Also, deleting a prefab asset that is used in a scene or another prefab should indeed emit a warning. We will look into that.

**Comment by EmptyKeys** [comment]
I see. I'm using prefabs as run-time objects to spawn stuff so marking everything as root felt weird. So I created simple component, which keeps references to prefabs, which I use in that scene. That solves both building and GameStudio warning issue.

I thought that would be actually proper solution.

You are right about loading, thought that all root stuff is loaded on start.

**Comment by xen2** [comment]
Yes, creating a simple script that references the asset (prefabs and others too) solves several problem at once:
* No need to mark it as root asset anymore
* No "asset missing" if you rename it (it's a hard reference that gets compile-time checked instead of a string)
* It gets loaded at startup (we'll probably need some kind of wrapper like `DeferredAsset<Prefab>` later to still allow async loading -- might come together with the scene streaming)

We use this pattern quite a lot too. We should probably present and recommend it in the documentation.

**Comment by boriscallens** [comment]
Maybe an explicit entity type would help discoverability for this? 
I realise this sounds trivial, but I feel that if this is a recommended pattern, it deserves explicit support?
Like a "PreloadEntitiesList" or something (naming is hard).
